File: backend/routes/health.py
# backend/routes/health.py
from fastapi import APIRouter, HTTPException
from services.health_service import HealthService
from models.schemas import HealthCheckResponse, ErrorResponse

router = APIRouter()
health_service = HealthService()

@router.get("/health/database", response_model=HealthCheckResponse)
def check_database_health():
    """
    Test database connection
    Returns server version and current time if connected
    """
    try:
        result = health_service.get_database_health()

        if result["database_status"] == "error":
            raise HTTPException(
                status_code=500,
                detail=result["details"].get("error", "Unknown database error")
            )

        return HealthCheckResponse(**result)

    except Exception as e:
        raise HTTPException(
            status_code=500,
            detail=f"Health check failed: {str(e)}"
        )

@router.get("/health/api")
def check_api_health():
    """Check API health (always returns 200 if this endpoint is reachable)"""
    return {
        "status": "healthy",
        "message": "API is running"
    }

File: backend-go/routes/health.go
package routes

import (
	"github.com/gofiber/fiber/v2"

	"stock-api/models"
	"stock-api/repositories"
	"stock-api/services"
)

// RegisterHealth wires the health-check endpoints onto the application.
func RegisterHealth(app *fiber.App, repo *repositories.HealthRepository) {
	svc := services.NewHealthService(repo)
	app.Get("/api/health/api", healthAPIHandler)
	app.Get("/api/health/database", healthDatabaseHandler(svc))
}

// healthAPIHandler godoc
// @Summary     Santé de l'API
// @Description Retourne 200 si le processus API est actif (ne nécessite pas SQL Server)
// @Tags        health
// @Produce     json
// @Success     200 {object} map[string]string "status=healthy"
// @Router      /api/health/api [get]
func healthAPIHandler(c *fiber.Ctx) error {
	return c.JSON(fiber.Map{
		"status":  "healthy",
		"message": "API is running",
	})
}

// healthDatabaseHandler godoc
// @Summary     Connexion SQL Server
// @Description Teste la connexion à SQL Server et retourne la version et l'heure serveur
// @Tags        health
// @Produce     json
// @Success     200 {object} models.HealthResponse
// @Failure     500 {object} map[string]string
// @Router      /api/health/database [get]
func healthDatabaseHandler(svc *services.HealthService) fiber.Handler {
	return func(c *fiber.Ctx) error {
		result, err := svc.GetDatabaseHealth()
		if err != nil {
			return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
				"status":  "error",
				"message": "Health check failed",
				"detail":  err.Error(),
			})
		}
		return c.Status(fiber.StatusOK).JSON(result)
	}
}

// ensure models is used for swag doc generation
var _ models.HealthResponse
